ably/ably_client.go

```diff
@@ -1,12 +1,25 @@
 package main
 
 import (
+	"time"
+
 	"github.com/ably/ably-go/ably"
 )
 
-func newAblyClient(testConfig TestConfig) (*ably.RealtimeClient, error) {
+// newAblyClient creates a new Ably realtime client, making multiple attempts
+// to avoid transient connection errors being reported as failures.
+func newAblyClient(testConfig TestConfig) (client *ably.RealtimeClient, err error) {
 	options := ably.NewClientOptions(testConfig.APIKey)
 	options.Environment = testConfig.Env
 
-	return ably.NewRealtimeClient(options)
+	timeout := 30 * time.Second
+	delay := 100 * time.Millisecond
+	for start := time.Now(); time.Since(start) < timeout; time.Sleep(delay) {
+		client, err = ably.NewRealtimeClient(options)
+		if err == nil {
```

Comment:
Shouldn't there be an attempt here to bail out if the error is not retriable? Eg an error in the given `ClientOptions` ?

Reply:
Perhaps, but I'm not sure how we would detect that here, and it will be pretty obvious in Locust if we've got the options wrong as no connections will be established at all, so I've deferred doing this properly to when we do it in ably-go: d5ee561